In [3]:
import pandas as pd

df = pd.read_json('streaming_users_dirty.json')

df.head(10)

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1
5,10005,20,Básico,670.2,Uruguay,Drama,2020-07-03,2
6,10006,37,Básico,346.6,Perú,Thriller,2019-07-26,1
7,10007,31,Estándar,974.6,Chile,Acción,2019-02-24,1
8,10008,36,Premium,1432.2,Colombia,Romance,2025-08-03,2
9,10009,37,Estándar,1375.4,Argentina,Thriller,2024-02-12,1


In [4]:
df.shape

(8160, 8)

In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   str    
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   str    
 5   favorite_genre            7920 non-null   str    
 6   last_login_date           7840 non-null   str    
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), str(4)
memory usage: 510.1 KB


In [6]:
df.isnull().sum()

user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64

In [15]:
df.describe()

,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.000000,8160.000000,7967.000000,8160.000000
mean,13995.433824,34.096814,1107.346153,1.800980
std,2310.810660,14.511304,5310.442604,11.334969
min,10000.000000,-5.000000,-120.000000,-1.000000
25%,11987.750000,25.000000,489.200000,0.000000
50%,13998.500000,33.000000,757.400000,1.000000
75%,15997.250000,42.000000,1045.700000,1.000000
max,17999.000000,150.000000,99999.000000,150.000000


Diccionario

►user_id = ID de usuario
►age = Edad
►suscription_plan = Plan de suscription
►monthly_watch_time_mins = Tiempo de visualizacion al mes (en minutos)
►country = País
►favorite_genre = Genero favorito
►last_login_late = Ultimo inicio de sesion
►customer_support_tickets = Registro digital generado cada vez que un cliente se comunica (pregunta, queja o solicitud)


Observaciones 
►El dataset cuenta con:
    ♣8160 filas
    ♣8 columnas
►"last_login_late" no tiene el formato que deberia tener, esta en str y deberia estar en "date" o "datetime"
►"monthly_watch_time_mins", "favorite_genre", y "last_login_late" suman un total de 753 nulos.

Valores imposibles

► "age","monthly_watch_time_wins", "customer_support_tickets"
    ♣los valores minimos son irreales ya que son negativos
    ♣los valores maximos son completamentes imposibles siendo tan altos (age = 150), (monthly_watch_time_wins = 99999 minutos)
    ♣Mientras el valor maximo de "customer_support_tickets" podrias ser factible, ya que es solo de 150